### PR TITLE
{math}[foss/2019a,intel/2019a] MUMPS v5.2.1

### PR DESCRIPTION
**Pull request by lexming**
(created using `eb --new-pr`)
Update of MUMPS to version 5.2.1 for 2019a toolchains. The main difference with previous builds are two patches that enable building its shared libs. The shared libs are installed alongside the static libs, so there should be no regression with respect previous easyconfigs.

### CODE REVIEW

**Comment by lexming** [comment]
Version bump to 5.2.1

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
nic20 - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/4b987eec93aab4fc2e27bbb3e947438b for a full test report.

File: easybuild/easyconfigs/m/MUMPS/MUMPS-5.2.1_shared-mumps.patch [hunk 1]
```diff
@@ -0,0 +1,65 @@
+source: https://src.fedoraproject.org/rpms/MUMPS/blob/master/f/MUMPS-shared.patch
```
Comment (smoors) [inline easybuild/easyconfigs/m/MUMPS/MUMPS-5.2.1_shared-mumps.patch]: @lexming please add Author and update the checksum?

File: easybuild/easyconfigs/m/MUMPS/MUMPS-5.2.1_shared-pord.patch [hunk 2]
```diff
@@ -0,0 +1,79 @@
+source: https://src.fedoraproject.org/rpms/MUMPS/blob/master/f/MUMPS-shared-pord.patch
```
Comment (smoors) [inline easybuild/easyconfigs/m/MUMPS/MUMPS-5.2.1_shared-pord.patch]: @lexming please add Author and update the checksum?

File: easybuild/easyconfigs/m/MUMPS/MUMPS-5.2.1_shared-mumps.patch (hunk 1 above)
Comment (lexming) [reply easybuild/easyconfigs/m/MUMPS/MUMPS-5.2.1_shared-mumps.patch]: Done :+1:

File: easybuild/easyconfigs/m/MUMPS/MUMPS-5.2.1_shared-pord.patch (hunk 2 above)
Comment (lexming) [reply easybuild/easyconfigs/m/MUMPS/MUMPS-5.2.1_shared-pord.patch]: Done :+1:

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
nic425 - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/212de580feb5898c7cd02aa2fb617151 for a full test report.

**Comment by smoors** [comment]
Going in, thanks @lexming!

**Pull request merged**